code/greedy_algorithms/src/huffman_coding/huffman_coding_encoding.java

```diff
@@ -0,0 +1,193 @@
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.util.PriorityQueue;
+import java.util.Scanner;
+import java.util.TreeMap;
+
+/* Huffman coding , decoding */
+class Huffman {
+    static final boolean readFromFile = false;
+    static final boolean newTextBasedOnOldOne = false;
+    static PriorityQueue<Node> nodes = new PriorityQueue<>((o1, o2) -> (o1.value < o2.value) ? -1 : 1);
+    static TreeMap<Character, String> codes = new TreeMap<>();
+    static String text = "";
+    static String encoded = "";
+    static String decoded = "";
+    static int ASCII[] = new int[128];
+
+    public static void main(String[] args) throws FileNotFoundException {
+        Scanner scanner = (readFromFile) ? new Scanner(new File("input.txt")) : new Scanner(System.in);
+        int decision = 1;
+        while (decision != -1) {
+            if (handlingDecision(scanner, decision)) continue;
+            decision = consoleMenu(scanner);
+        }
+    }
+
+    private static int consoleMenu(Scanner scanner) {
+        System.out.println("\n---- Menu ----\n" +
+                "-- [-1] to exit \n" +
+                "-- [1] to enter new text\n" +
+                "-- [2] to encode a text\n" +
+                "-- [3] to decode a text");
+        int decision = Integer.parseInt(scanner.nextLine());
+        if (readFromFile)
+            System.out.println("Decision: " + decision + "\n---- End of Menu ----\n");
+        return decision;
+    }
+
+    private static boolean handlingDecision(Scanner scanner, int decision) {
+        if (decision == 1) {
+            if (handleNewText(scanner)) return true;
+        } else if (decision == 2) {
+            if (handleEncodingNewText(scanner)) return true;
+        } else if (decision == 3) {
+            handleDecodingNewText(scanner);
```

Comment:
I really don't like the inconsistency that's there that the encoding function returns a boolean, while the decoding one doesn't.